packages/babel-plugin-transform-void-to-nothing/src/index.js

```diff
@@ -0,0 +1,22 @@
+"use strict";
+
+const v0 = require("../../babel-helper-is-void-0/src");
+
+module.exports = function({ types: t }) {
+  // "var a = void 0;" -> "var a;"
+  const isVoid0 = v0(t);
+  return {
+    name: "transform-void-to-nothing",
+    visitor: {
+      VariableDeclaration({node: {kind: kind, declarations: declarations}}) {
+        if (kind !== "const") {
+          for (const declaration of declarations) {
+            if (isVoid0(declaration.init)) {
```

Comment:
I think we should just check for `undefined` value. `void 0` is just one representation of that value so no need to limit ourselves here (or expect other plugins to kick in before). 

I think we should also account for other places, not just declarations. `return undefined` is currently not transformed to just `return`. 

There's also call expressions `foo(x)` where we can confidently determine that `x` is `undefined`. The only issue here is that removing argument has a side-effect — arguments.length is no longer the same. That would have to be marked as an unsafe transform I guess.

/cc @hzoo @danez